packages/cli/src/commands/dev.js

```diff
@@ -9,26 +9,36 @@ export default {
     ...common,
     ...server
   },
-  async run(cmd) {
-    const argv = cmd.getArgv()
+  async run() {
+    this.disableForceExit()
+
+    const argv = this.getArgv()
+
+    const getConfig = () => {
+      return this.getNuxtConfig(argv, { dev: true })
+    }
 
     const errorHandler = (err, instance) => {
       instance && instance.builder.watchServer()
       consola.error(err)
     }
 
+    const config = await getConfig()
+
+    if (argv.lock) {
+      await this.lock(config.srcDir || config.rootDir, { autoUnlock: false })
+    }
+
     // Start dev
-    async function startDev(oldInstance) {
+    const startDev = async (config, oldInstance) => {
```

Comment:
Why changed to arrow function? named functions help to debug easier

Reply:
When using an arrow function we can use `this.` to access `NuxtCommand` instead of having to use the first parameter of the run method. At first I wanted to make `startDev` a class method as well, ~~but there is no mechanism~~ wait while writing this I realised because of the from method you implemented we can just have `dev` export a NuxtCommand instance and make startDev a class method. What do you think about that?

Reply:
I see your reasons. I think keeping it here would be better to keep NuxtCommand as simple as possible.